src/firstvoices/backend/tests/test_models/test_character_model.py

```diff
@@ -0,0 +1,177 @@
+
+    @pytest.mark.django_db
+    def test_clean_confusables_basic(self):
+        """Apply confusable transducer as defined in alphabet"""
+        alphabet = AlphabetFactory.create(
+            input_to_canonical_map=[
+                {"in": "á", "out": "a"},
+                {"in": "ᐱ", "out": "A"},
+                {"in": "_b", "out": "b"},
+                {"in": "č", "out": "cv"},
+            ],
+        )
+        assert alphabet.clean_confusables("X") == "X"
+        assert alphabet.clean_confusables("á") == "a"
+        assert alphabet.clean_confusables("_b") == "b"
+        assert alphabet.clean_confusables("č") == "cv"
+        assert alphabet.clean_confusables("áx ᐱᐱᐱ") == "ax AAA"
+        assert alphabet.clean_confusables("aᐱᐱx _bč") == "aAAx bcv"
+        IgnoredCharacterFactory(site=alphabet.site, title="/")
+        assert alphabet.clean_confusables("ᐱ/_/b/č") == "A/_/b/cv"
+
+    @pytest.mark.django_db
+    def test_clean_confusables_no_feeding(self):
+        """Default confusables transducer does not allow multi-rule application"""
+        alphabet = AlphabetFactory.create(
+            input_to_canonical_map=[
+                {"in": "AA", "out": "A"},
+                {"in": "A", "out": "a"},
+                {"in": "č", "out": "c"},
+                {"in": "čh", "out": "čh"},
+            ],
+        )
+        assert alphabet.clean_confusables("A") == "a"
+        assert alphabet.clean_confusables("AA") == "A"
+        assert alphabet.clean_confusables("AAA") == "Aa"
+        assert alphabet.clean_confusables("č") == "c"
+        assert alphabet.clean_confusables("čh") == "čh"
+        IgnoredCharacterFactory(site=alphabet.site, title="/")
+        assert alphabet.clean_confusables("A/A") == "a/a"
+
+    @pytest.mark.skip("g2p with regex escape ignores mapping with escaped in-chars")
+    @pytest.mark.django_db
+    def test_clean_confusables_regex_escape(self):
+        """Default confusables transducer ignores regex rules"""
+        alphabet = AlphabetFactory.create(
+            input_to_canonical_map=[
+                {"in": r"^", "out": "A"},
+                {"in": r".", "out": "o"},
+                {"in": r"ng.", "out": "ng"},
+                {"in": r"\d", "out": "D"},
+            ],
+        )
+        assert alphabet.clean_confusables("test") == "test"
+        assert alphabet.clean_confusables(r"^^a") == "AAa"
+        assert alphabet.clean_confusables(r"test.") == "testo"
+        assert alphabet.clean_confusables(r"testing.") == "testing"
+        assert alphabet.clean_confusables(r"\d\d") == "DD"
+
+
+class TestCharacterModel:
+    @pytest.mark.django_db
+    def test_characters_same_name(self):
+        """Character can't be created with the same name as another"""
+        char = CharacterFactory.create()
+        with pytest.raises(IntegrityError):
+            CharacterFactory.create(site=char.site, title=char.title)
+
+    @pytest.mark.django_db
+    def test_character_variant_same_name(self):
```

Comment:
This just a note for the future but if you want to easily test all combos of these factories, the test parameter feature lets you multiple two lists together